### PR TITLE
feat: router write line protocol duration

### LINKED ISSUE #5506: Observe write line-protocol duration
Now, a request process will go through `Line-Protocol` parse, dml-handlers processing,etc. 

image

I would like to be able to observe write line-protocol stage duration.

The metrics 
name: `http_write_lines_duration`  
type: `DurationHistogram`
attr: `namespace`

### PR DESCRIPTION
**Pull request by aierui**
Closes #5506 

Improving write line-protocol stage observability.

- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: router/src/server/http.rs [hunk 1]
```diff
@@ -361,6 +369,11 @@ where
         };
 
         let num_tables = batches.len();
+        let duration = start_instant.elapsed();
+        let attributes = Attributes::from([("namespace", format!("{}", namespace).into())]);
```
Comment (domodwyer) [inline router/src/server/http.rs:373]: Unfortunately labelling with the namespace will produce very high cardinality metrics (1 per namespace) - most metric systems perform poorly when this happens :(

This means we can't label with `namespace` I'm afraid - we'll switch our alerting system to IOx soon enough and high cardinality workloads will be better!

**Review by domodwyer — COMMENTED** [review]
Thanks @aierui :heart:

Code looks great, just minor comments :) Would be great to merge this!

File: router/src/server/http.rs [hunk 2]
```diff
@@ -276,13 +278,18 @@ impl<D> HttpDelegate<D, SystemProvider> {
                 "number of HTTP requests rejected due to exceeding parallel request limit",
             )
             .recorder(&[]);
+        let http_write_lines_duration = metrics.register_metric::<DurationHistogram>(
+            "http_write_lines_duration",
+            "write latency of line protocol lines",
+        );
```
[the remark this answers is not in the document]
Comment (aierui) [reply router/src/server/http.rs:284]: Thank you for your careful guidance, I will adjust the variable name.

File: router/src/server/http.rs (hunk 1 above)
Comment (aierui) [reply router/src/server/http.rs:373]: I see what you mean, I'll remove this label.

**Review by domodwyer — APPROVED** [review]
Nice! Thanks for the help @aierui :)

Insight to a system is always valuable!

**Pull request merged**